src/middlewared/middlewared/auth.py

```diff
@@ -38,12 +44,56 @@ def dump(self):
 
 
 class UserSessionManagerCredentials(SessionManagerCredentials):
-    def __init__(self, user):
+    """ Credentials for authenticated user session """
+
+    def __init__(self, user: dict, assurance: AuthenticatorAssuranceLevel | None):
+        """
+        user: dictionary generated by `auth.authenticate_user`
+
+        assurance: authenticator assurance level for the session. This
+        may be None if the session is authenticated by virtue of unix domain
+        socket connection.
+
+        Our default is AAL1 which allows authenticated sessions to use
+        credentials for up to 30 days before requiring reauthentication, and
+        does not set an inactivity timeout.
+
+        See comments in utils/auth.py for more information.
+        """
+        now = monotonic()
         self.user = user
+        self.assurance = assurance
         self.allowlist = Allowlist(user["privilege"]["allowlist"])
         self.is_user_session = True
+        self.login_at = datetime.now(UTC)
+        self.expiry = None
+        self.inactivity_timeout = None
```

Comment:
Is there a reason why these are not being initialized thusly:
```
        self.expiry = now + AA_LEVEL1.max_session_age
        self.inactivity_timeout = AA_LEVEL1.max_inactivity
```

Reply:
Yes. This is also used with connections via Unix Domain Socket (in which case assurance is None).